harness/determined/cli/workspace.py

```diff
@@ -171,21 +170,36 @@ def list_workspace_namespace_bindings(args: argparse.Namespace) -> None:
 
 def set_workspace_namespace_binding(args: argparse.Namespace) -> None:
     sess = cli.setup_session(args)
-    if args.cluster_name and not args.namespace:
-        raise api.errors.BadRequestException("must provide --namespace NAMESPACE")
+    if args.cluster_name and not (
+        args.namespace or args.auto_create_namespace or args.auto_create_namespace_all_clusters
+    ):
+        raise api.errors.BadRequestException(
+            "must provide --namespace NAMESPACE, "
+            + "--auto-create-namespace, or --auto-create-namespace-all-clusters"
```

Comment:
```
HPE-Q5V7VPFH71:determined jgong$ det w bindings set myss --cluster-name name
Failed to set workspace-namespace binding: must provide --namespace NAMESPACE, --auto-create-namespace, or --auto-create-namespace-all-clusters
```
The error message might be a little confusing because we don't allow using `det w bindings set myws --cluster-name name --auto-create-namespace-all-clusters`.

Reply:
Changed the  error message to "must provide --namespace NAMESPACE or --auto-creeate-namespace, or remove --cluster-name CLUSTER_NAME and specify --auto-create-namespace-all-clusters"